static/js/message_edit.js

```diff
@@ -185,6 +186,44 @@ export function get_deletability(message) {
     return false;
 }
 
+export function stream_and_topic_exist_in_edit_history(message, stream_id, topic) {
+    /*  Checks to see if a stream_id and a topic match any historical
+        stream_id and topic state in the message's edit history.
+
+        Does not check the message's current stream_id and topic for
+        a match to the stream_id and topic parameters.
+     */
+    const narrow_dict = {stream_id, topic};
+    const message_dict = {stream_id: message.stream_id, topic: message.topic};
+
+    if (!message.edit_history) {
+        return false;
+    }
+
+    for (const edit_history_event of message.edit_history) {
+        if (!edit_history_event.prev_stream && !edit_history_event.prev_topic) {
+            continue;
+        }
+        if (
+            edit_history_event.prev_stream &&
+            edit_history_event.stream === message_dict.stream_id
+        ) {
+            message_dict.stream_id = edit_history_event.prev_stream;
+        }
+        if (
+            edit_history_event.prev_topic &&
+            util.lower_same(edit_history_event.topic, message_dict.topic)
```

Comment:
This is also an assertion of the correctness of the edit history data structure.